Functions/xASL_bids_CheckDatasetDescription.m

```diff
@@ -0,0 +1,86 @@
+function [bImportedExploreASL, bImportedSameVersion, versionExploreASLBIDS, bImportedBETA] = xASL_bids_CheckDatasetDescription(datasetDescription,versionExploreASL)
+%xASL_bids_CheckDatasetDescription Check the dataset_description.json field
+% contents with special regard to the import version.
+%
+% FORMAT: [bImportedExploreASL, bImportedSameVersion, versionExploreASLBIDS, bImportedBETA] = xASL_bids_CheckDatasetDescription(datasetDescription,versionExploreASL)
+% 
+% INPUT:
+%   datasetDescription - Structure which defines the dataset_description fields (STRUCT, REQUIRED)
+%
+% OUTPUT:
+%   bImportedExploreASL   - Was the dataset imported with ExploreASL or not (BOOLEAN)
+%   bImportedSameVersion  - True if the version matches exactly the current version (full match with BETA) (BOOLEAN)
+%   versionExploreASLBIDS - Empty (if above == false), or string with version (excluding the BETA) (STRING)
+%   bImportedBETA         - True if BETA version was used -> see above, otherwise false (BOOLEAN)
+%                         
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  Check the dataset_description.json field contents with special regard to the import version.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:      [bImportedExploreASL, bImportedSameVersion, versionExploreASLBIDS, bImportedBETA] = xASL_bids_CheckDatasetDescription(datasetDescription,x.Version);
+%               
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+
+    %% Input check
+    if nargin<1 || isempty(datasetDescription)
+        error('Please provide a struct based on the dataset_description.json BIDS file...');
+    end
+    if nargin<2 || isempty(versionExploreASL)
+        error('Please provide the version of your ExploreASL installation...');
+    end
+    
+    
+    %% Defaults
+    bImportedExploreASL = false;
+    bImportedSameVersion = false;
+    versionExploreASLBIDS = [];
+    bImportedBETA = false;
+    
+    
+    %% Check structure
+    
+    % Was the dataset imported with ExploreASL or not
+    if isfield(datasetDescription,'Acknowledgements') && ...
+       ~isempty(regexp(datasetDescription.Acknowledgements,'Imported with ExploreASL', 'once'))
+        bImportedExploreASL = true;
+    end
+    
+    % True if the version matches exactly the current version
+    if bImportedExploreASL
+        indexVersionString = regexp(datasetDescription.Acknowledgements,'Imported with ExploreASL', 'once')+length('Imported with ExploreASL');
+        fileVersion = datasetDescription.Acknowledgements(indexVersionString+1:end);
```

Comment:
Sorry. A wrong assumption that we have the only acknowledgement is from ExploreASL.

We have to go for "Imported with ExploreASL XX.YY.ZZ(_BETA)." Anything can come after the last "."

Reply:
Note that we previously didn't end "our" statement with a fullstop, but we probably should in case there are more to add there.

Reply:
then be careful to detect this last fullstop, but then remove it from the fileVersion

Reply:
Makes sense. I think this should work with 773a8c2 👍

Reply:
If it's a beta version, I search for that and use it to find the last version character. If not I just look for the third dot. That should work pretty robustly, as long as people don't write wrong version formats in there manually.